src/cdk/explorer/tree/treeInspector.ts

```diff
@@ -20,6 +20,20 @@ export function getTypeAttributeOrDefault(construct: ConstructTreeEntity, defaul
     return defaultValue
 }
 
+/**
+ * Determines the props of a construct if they exist
+ *
+ * @param construct CDK construct
+ */
+export function getProperties(construct: ConstructTreeEntity): ConstructProps | undefined {
+    const attributes = construct.attributes
+    if (attributes && attributes[CfnResourceKeys.PROPS]) {
+        return attributes[CfnResourceKeys.PROPS] as ConstructProps
+    }
+
+    return undefined
```

Comment:
If you returned an empty array here, calling code wouldn't need to make "is undefined" checks. Do you ever need to know that `construct` didn't have PROPS attributes?

Reply:
ah good call, I like that suggestion and we can iterate on the array directly. 

There will be other namespaces that represent properties (just not yet). This specific key is one that directly represents the properties of a CloudFormation resource. The idea is that higher level constructs (that are composed of multiple CloudFormation resources) may also share properties.

in a tree:

- high level construct
   - makePublic: true
   - CloudFormation resource
      - some property
          - makePublic: true
   - CloudFormation resource
      - visibility: PUBLIC  

In this example: a high level construct that CDK authors will interact with has a property that they set to true. That property then results in everything below it. Those are values that the high level construct has set on behalf of the user.

This helps to illustrate the impact of properties in underlying resources that are set based on the code that's written.

Reply:
ah actually the reason I went with a typed structure was to enclose it in a type. Since this may return other types based on the properties that are assembled together, I went with a type `ConstructProps` enclosure around a bag of properties. `props` is the key I used in that bag because that's a term that always means a construct's properties.

let me know if you still feel this should change though.